subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/CacheBackedTaskHistoryRepository.java

```diff
@@ -316,6 +320,7 @@ private static boolean isOutputEntry(String path, NormalizedFileSnapshot fileSna
             FileCollectionSnapshot result;
             try {
                 FileCollectionSnapshotter snapshotter = snapshotterRegistry.getSnapshotter(propertySpec.getSnapshotter());
+                LOGGER.debug("Snapshotting property '{}' for {}", propertySpec, task);
```

Comment:
Should we really add `'` around the property spec? Currently the output looks like this:
```
Snapshotting property 'rootSpec$1 (RELATIVE)' for task ':jar'
```